certbot/certbot/_internal/auth_handler.py

```diff
@@ -87,9 +87,34 @@ def handle_authorizations(self, orderr: messages.OrderResource,
 
                 # If debug is on, wait for user input before starting the verification process.
                 if config.debug_challenges:
+                    msg = []
+                    if config.verbose_count > 0:
+                        http01_achalls = {}
+                        dns01_achalls = {}
+                        for achall in achalls:
+                            if isinstance(achall.chall, challenges.HTTP01):
+                                http01_achalls[achall.chall.uri(achall.domain)] = (
+                                    achall.validation(achall.account_key) + "\n"
+                                )
+                            if isinstance(achall.chall, challenges.DNS01):
+                                dns01_achalls[achall.validation_domain_name(achall.domain)] = (
+                                    achall.validation(achall.account_key) + "\n"
+                                )
+                        if http01_achalls:
+                            msg.append("\nThe following URLs should be accessible from the world "
+                                       "wide web and return the value mentioned:\n")
+                            for uri, token in http01_achalls.items():
+                                msg.append(f"{uri}: {token}")
+                        if dns01_achalls:
+                            msg.append("\nThe following FQDNs should return a TXT resource "
+                                       "record with the value mentioned:\n")
+                            for fqdn, token in dns01_achalls.items():
+                                msg.append(f"{fqdn}: {token}")
```

Comment:
I think it's probably worth moving this block into its own function, which summarizes the challenges as a string.